src/System.Net.Requests/tests/HttpWebRequestTest.cs

```diff
@@ -521,10 +1051,13 @@ public void Abort_BeginGetRequestStreamThenAbort_EndGetRequestStreamThrowsWebExc
 
             request.BeginGetResponse(new AsyncCallback(RequestStreamCallback), state);
 
-            request.Abort();
-            Assert.Equal(1, state.RequestStreamCallbackCallCount);
-            WebException wex = state.SavedRequestStreamException as WebException;
-            Assert.Equal(WebExceptionStatus.RequestCanceled, wex.Status);
+            if (!request.HaveResponse)
```

Comment:
It's not saying they inherently deadlock.  It's saying you need to be aware of the behavior so that you don't write code that deadlocks, just being aware that Abort may cause the callback to be synchronously invoked, so you shouldn't write code with dependencies between that callback and the caller that would result in a deadlock.